model/value_float.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2013 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"strconv"
+)
+
+var (
+	// ZeroSamplePair is the pseudo zero-value of SamplePair used to signal a
+	// non-existing sample pair. It is a SamplePair with timestamp Earliest and
+	// value 0.0. Note that the natural zero value of SamplePair has a timestamp
+	// of 0, which is possible to appear in a real SamplePair and thus not
+	// suitable to signal a non-existing SamplePair.
+	ZeroSamplePair = SamplePair{Timestamp: Earliest}
+)
+
+// A SampleValue is a representation of a value for a given sample at a given
+// time.
+type SampleValue float64
```

Comment:
We could possibly do this and remove duplication methods. (and get the FloatString implementation to this file)

```suggestion
type SampleValue FloatString
```

Reply:
I tried to do this, but it didn't remove the need for duplicate methods as we still needed to implement the methods for SampleValue directly. I think this is because for the new histogram structs, we use FloatString as the type directly, but for the old float struct we alias it to SampleValue so it requires its own set of methods. If we wanted to remove the redundancy, we should remove the SampleValue alias and make the Sample use FloatString directly, but this seems like a very big change to existing code, and others importing this library probably already use SampleValue, including Mimir. Alternatively, we could remove FloatString and use SampleValue in its place, but then the naming would be confusing.